packages/vue-components/src/Modal.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <!-- NavBar has z-index of 1000, hence the z-index here should exceed that -->
+  <vue-final-modal
+    v-if="isMounted"
+    v-model="show"
+    ssr
```

Comment:
Just to check, was `ssr=false` tried?

Presumably it prevents the modal from getting rendered during the ssr cycles, but I have been getting errors with it (might be a bug over there). The `isMounted` way seems like a good alternative.

Should we set `ssr=false` for clarity? (shouldn't be rendered during ssr) (maybe with a comment as well)

https://github.com/vue-final/vue-final-modal/blob/master/lib/VueFinalModal.vue#L3

Reply:
Yup! When `isMounted` is used, both `ssr` and `ssr=false` work. (Without `isMounted`, both don't work.) I've added a comment to clarify.

Just to check my understanding, why should modals not be rendered during SSR cycles? I used browser devtools to check for SSR hydration issues, not sure if there's any other way to detect SSR issues.

Reply:
> EDIT: ssr=false breaks the snapshot tests, so I reverted the change

Ok, ugh, the issue seems to be that `$vfm` (being a singleton) isn't really testable. Wonder if we should transfer this once we have e2e tests setup in the other PR. @jonahtanjz  🤔

Ok with snapshot tests now though. Let's update the comment as appropriate too.

> Just to check my understanding, why should modals not be rendered during SSR cycles? I used browser devtools to check for SSR hydration issues, not sure if there's any other way to detect SSR issues.

That depends, in MarkBind's case we've decided to discard modals from indexing, per the updates you did in `algolia.js`.

For this implementation https://github.com/vue-final/vue-final-modal/blob/master/lib/VueFinalModal.vue#L3, the `ssr` option seems to render a `display: none` modal. This might be useful if you want to include modal content inside search engine results.

Another case is if you want to show the modal on-page-load (rarely, since popups are triggered by user action usually), you would need to render it during ssr as well. (by setting the initial value of `show=true`)

Reply:
> per the updates you did in algolia.js.

EDIT on this: sorry, just realised with this discussion these selectors have to be retargeted / no longer necessary for SSR, something we missed out in #1534. (same rationale of excluding from indexing still applies though)

I've added it on to #1815.

Reply:
I see, thank you so much for the explanation!
Have updated the comment too :)

Reply:
> Wonder if we should transfer this once we have e2e tests setup in the other PR. @jonahtanjz 🤔

I think we can keep the current snapshot tests for now since the e2e tests are focused on the live preview part (this may not be relevant there?). We can consider moving this once we have more e2e tests for the other frontend components as well.